src/pages/SigninPage.js

```diff
@@ -0,0 +1,60 @@
+import React, { useState, useEffect } from 'react';
+import LoginForm from '../components/Signin';
+import { useDispatch, useSelector } from 'react-redux';
+import { loggedUser } from '../redux/actionCreators/signinAction';
+import { NavigationBar } from '../components/Nav/NavigationBar';
+import { Footer } from '../components/Footer/Footer';
+import { useTranslation } from 'react-i18next';
+
+const signin = () => {
+  const [errors, setErrors] = useState('');
+  const [details, setDetails] = useState({ email: '', password: '' });
+  const dispatch = useDispatch();
+  const { loading, isLoggedIn, error } = useSelector((state) => state.auth);
+  const { t } = useTranslation();
+
+  const submitHandler = (e) => {
+    e.preventDefault();
+    const { email, password } = details;
+    const emailFormat = /^[a-z][\w-\.]*\@[a-z]+\.[a-z]{2,4}$/;
+    const passwordFormat = /[a-z]/;
+
+    if (!emailFormat.test(email)) {
+      setErrors(t('signin.errorEmail'));
+      setDetails({ ...details, password: '' });
+    } else if (!password) {
+      setErrors(t('signin.errorPassword'));
+      setDetails({ ...details, password: '' });
+    } else if (!passwordFormat.test(password)) {
+      setErrors(t('signin.errorPass'));
+      setDetails({ ...details, password: '' });
+    } else {
+      dispatch(loggedUser(details));
+    }
+  };
```

Comment:
We already have these validations from the backend. Can you explain why we are doing the same thing here?

Reply:
its because the validations from the backend take some time to respond, so I thought to have some simple form validations that will be familiar to the user

Reply:
its because the validations from the backend take some time to respond, so I thought to have some simple form validations that will be familiar to the user